src/kernel.cr

```diff
@@ -68,8 +68,224 @@ def printf(format_string, args : Array | Tuple) : Nil
+#            | (numeric fmt) | is used for negative numbers formatted as
+#            |               | complements.
+# ```
+#
+# Examples of flags:
+#
+# Decimal number conversion
+# ```
+# sprintf "%d", 123  # => "123"
+# sprintf "%+d", 123 # => "+123"
+# sprintf "% d", 123 # => " 123"
+# ```
+#
+# Octal number conversion
+# ```
+# sprintf "%o", 123   # => "173"
+# sprintf "%+o", 123  # => "+173"
+# sprintf "%o", -123  # => "-173"
+# sprintf "%+o", -123 # => "-173"
+# ```
+#
+# Hexadecimal number conversion
+# ```
+# sprintf "%x", 123   # => "7b"
+# sprintf "%+x", 123  # => "+7b"
+# sprintf "%x", -123  # => "-7b"
+# sprintf "%+x", -123 # => "-7b"
+# sprintf "%#x", 0    # => "0"
+# sprintf "% x", 123  # => " 7b"
+# sprintf "% x", -123 # => "-7b"
+# sprintf "%X", 123   # => "7B"
+# sprintf "%#X", -123 # => "-7B"
+# ```
+#
+# Binary number conversion
+# ```
+# sprintf "%b", 123    # => "1111011"
+# sprintf "%+b", 123   # => "+1111011"
+# sprintf "%+b", -123  # => "-1111011"
+# sprintf "%b", -123   # => "-1111011"
+# sprintf "%#b", 0     # => "0"
+# sprintf "% b", 123   # => " 1111011"
+# sprintf "%+ b", 123  # => "+ 1111011"
+# sprintf "% b", -123  # => "-1111011"
+# sprintf "%+ b", -123 # => "-1111011"
+# ```
+#
+# Floating point conversion
+# ```
+# sprintf "%a", 123 # => "0x1.ecp+6"
+# sprintf "%A", 123 # => "0X1.ECP+6"
+# ```
+#
+# Exponential form conversion
+# ```
+# sprintf "%g", 123.4          # => "123.4"
+# sprintf "%g", 123.4567       # => "123.457"
+# sprintf "%20.8g", 1234.56789 # => "           1234.5679"
+# sprintf "%20.8g", 123456789  # => "       1.2345679e+08"
+# sprintf "%20.8G", 123456789  # => "       1.2345679E+08"
+# sprintf "%20.8g", -123456789 # => "       -1.2345679e+08"
+# sprintf "%20.8G", -123456789 # => "       -1.2345679E+08"
+# ```
+#
+# The field width is an optional integer, followed optionally by a
+# period and a precision.  The width specifies the minimum number of
+# characters that will be written to the result for this field.
+#
+# Examples of width:
+# ```
+# sprintf "%20d", 123   # => "                 123"
+# sprintf "%+20d", 123  # => "                +123"
+# sprintf "%020d", 123  # => "00000000000000000123"
+# sprintf "%+020d", 12  # => "+0000000000000000123"
+# sprintf "% 020d", 123 # => " 0000000000000000123"
+# sprintf "%-20d", 123  # => "123                 "
+# sprintf "%-+20d", 123 # => "+123                "
+# sprintf "%- 20d", 123 # => " 123                "
+# sprintf "%020x", -123 # => "00000000000000000-7b"
+# sprintf "%020X", -123 # => "00000000000000000-7B"
+# ```
+#
+# For numeric fields, the precision controls the number of decimal places
+# displayed.  For string fields, the precision determines the maximum
+# number of characters to be copied from the string.  (Thus, the format
+# sequence <code>%10.10s</code> will always contribute exactly ten
+# characters to the result.)
+#
+# Examples of precisions:
+#
+# precision for `d`, `o`, `x` and `b` is
```

Comment:
This sentence should be properly formatted (start with capital letter, period at the end). Ditto for the following.